src/traces/surface/attributes.js

```diff
@@ -95,6 +95,13 @@ module.exports = {
         valType: 'data_array',
         description: 'Sets the text elements associated with each z value.'
     },
+    intensity: {
+        valType: 'data_array',
+        description: [
+            'Sets the surface intensity values,',
+            'used for setting a color scale independent of z'
+        ].join(' ')
+    },
     zauto: colorscaleAttrs.zauto,
     zmin: colorscaleAttrs.zmin,
     zmax: colorscaleAttrs.zmax,
```

Comment:
Ok, a couple things will need to happen here:

- rename `intensity` -> `surfacecolor`
- add `cauto`, `cmin` and `cmax` (which will take up the role of `zauto`, `zmin` and `zmax` respectively)
- move `zauto`, `zmin` and `zmax` to the bottom of the attribute object in a `_deprecated` block (see example [here](https://github.com/plotly/plotly.js/blob/dd1547992013505889ac6ff3294546db79c9f882/src/components/errorbars/attributes.js#L130-L139))
- add a backward-compatible block in [`cleanData`](https://github.com/plotly/plotly.js/blob/dd1547992013505889ac6ff3294546db79c9f882/src/plot_api/plot_api.js#L764):

```js
if(trace.type === 'surface') {
  if(trace.zauto) trace.cauto = trace.zauto; 
  delete trace.zauto

  /* and so on for 'zmin' and 'zmax'
}
```